js/apps/admin-ui/src/identity-providers/add/DiscoverySettings.tsx

```diff
@@ -50,152 +33,92 @@ const Fields = ({ readOnly }: DiscoverySettingsProps) => {
 
   return (
     <div className="pf-c-form pf-m-horizontal">
-      <FormGroup
+      <TextControl
+        name="config.authorizationUrl"
         label={t("authorizationUrl")}
-        fieldId="kc-authorization-url"
-        isRequired
-        validated={
-          errors.config?.authorizationUrl
-            ? ValidatedOptions.error
-            : ValidatedOptions.default
-        }
-        helperTextInvalid={t("required")}
-      >
-        <KeycloakTextInput
-          type="url"
-          data-testid="authorizationUrl"
-          id="kc-authorization-url"
-          validated={
-            errors.config?.authorizationUrl
-              ? ValidatedOptions.error
-              : ValidatedOptions.default
-          }
-          isReadOnly={readOnly}
-          {...register("config.authorizationUrl", { required: true })}
-        />
-      </FormGroup>
-
-      <FormGroup
+        type="url"
+        readOnly={readOnly}
+        rules={{
+          required: t("required"),
+        }}
+      />
+      <TextControl
+        name="config.tokenUrl"
         label={t("tokenUrl")}
-        fieldId="tokenUrl"
-        isRequired
-        validated={
-          errors.config?.tokenUrl
-            ? ValidatedOptions.error
-            : ValidatedOptions.default
-        }
-        helperTextInvalid={t("required")}
-      >
-        <KeycloakTextInput
-          type="url"
-          id="tokenUrl"
-          data-testid="tokenUrl"
-          validated={
-            errors.config?.tokenUrl
-              ? ValidatedOptions.error
-              : ValidatedOptions.default
-          }
-          isReadOnly={readOnly}
-          {...register("config.tokenUrl", { required: true })}
-        />
-      </FormGroup>
-      <TextField
-        field="config.logoutUrl"
-        label="logoutUrl"
-        isReadOnly={readOnly}
+        type="url"
+        readOnly={readOnly}
+        rules={{
+          required: t("required"),
+        }}
+      />
+      <TextControl
+        name="config.logoutUrl"
+        label={t("logoutUrl")}
+        readOnly={readOnly}
       />
-      <TextField
-        field="config.userInfoUrl"
-        label="userInfoUrl"
-        isReadOnly={readOnly}
+      <TextControl
+        name="config.userInfoUrl"
+        label={t("userInfoUrl")}
+        readOnly={readOnly}
       />
-      <TextField field="config.issuer" label="issuer" isReadOnly={readOnly} />
-      <SwitchField
-        field="config.validateSignature"
-        label="validateSignature"
-        isReadOnly={readOnly}
+      <TextControl
+        name="config.issuer"
+        label={t("issuer")}
+        readOnly={readOnly}
+      />
+      <DefaultSwitchControl
+        name="config.validateSignature"
+        label={t("validateSignature")}
+        isDisabled={readOnly}
```

Comment:
```suggestion
        readOnly={readOnly}
```

Reply:
readonly doesn't work on switches

Reply:
Ah yeah, I see that now. Thanks!

Reply:
the switch field did `isDisabled={isReadOnly}`